File: app/controllers/edit_category_controller.rb
class EditCategoryController < UITableViewController
  extend IB

  attr_accessor :category

  outlet :saveButton, UIBarButtonItem

  def viewDidLoad
    super

    @colors = Category::COLORS
    @category_name = @category.name
    @category_color_index = @category.colorIndex
  end

  def save
    self.view.endEditing(true)

    @category.name = @category_name
    @category.colorIndex = @category_color_index
    FkP.save

    self.navigationController.popViewControllerAnimated(true)
  end

  def cancel
    # do nothing, edited variables will simply be deleted

    self.navigationController.popViewControllerAnimated(true)
  end

  def nameDidChange(sender)
    @category_name = sender.text
  end

  def hideKeyboard
    self.view.endEditing(true)
  end


  #### text field delegate methods ####

  def textFieldShouldReturn(textField)
    textField.resignFirstResponder
  end


  #### table view delegate methods ####

  def numberOfSectionsInTableView(tableView)
    2
  end

  def tableView(tableView, numberOfRowsInSection: section)
    case section
    when 0
      1
    when 1
      @colors.count
    end
  end

  def tableView(tableView, cellForRowAtIndexPath: indexPath)
    case indexPath.section
    when 0
      cell = tableView.dequeueReusableCellWithIdentifier("CategoryNameCell")
      cell ||= UITableViewCell.alloc.initWithStyle(UITableViewCellStyleDefault, reuseIdentifier: "CategoryNameCell")
      cell.nameTextField.delegate = self
      cell.nameTextField.text = @category_name
      cell
    when 1
      cell = tableView.dequeueReusableCellWithIdentifier("CategoryColorCell")
      cell ||= UITableViewCell.alloc.initWithStyle(UITableViewCellStyleDefault, reuseIdentifier: "CategoryColorCell")
      
      color = @colors[indexPath.row]
      if @category_color_index == indexPath.row
        cell.accessoryType = UITableViewCellAccessoryCheckmark
        @selectedIndexPath = indexPath
      end

      cell.colorNameLabel.text = color[:name]
      cell.colorMark.color = color[:value]
      cell
    end
  end

  def tableView(tableView, didSelectRowAtIndexPath: indexPath)
    if indexPath.section == 1
      tableView.cellForRowAtIndexPath(@selectedIndexPath).accessoryType = UITableViewCellAccessoryNone

      @category_color_index = indexPath.row
      tableView.cellForRowAtIndexPath(indexPath).accessoryType = UITableViewCellAccessoryCheckmark
      @selectedIndexPath = indexPath

      tableView.deselectRowAtIndexPath(indexPath, animated: true)
    end
  end

end

File: Forking Pasta/EditCategoryController.swift
import UIKit
import KingPastaKit

class EditTimeZoneController: UITableViewController {
    
    var timeZone: TimeZone!
    
    private var colors = TimeZone.colors
    private var timeZoneName: String?
    private var timeZoneColorIndex: Int?
    private var selectedIndexPath: NSIndexPath?
    
    @IBOutlet weak var saveButton: UIBarButtonItem!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        timeZoneName = timeZone.name
        timeZoneColorIndex = timeZone.colorIndex
    }
    
    func save() {
        view.endEditing(true)
        
        try! FkP.realm.write {
            self.timeZone.name = self.timeZoneName ?? ""
            self.timeZone.colorIndex = self.timeZoneColorIndex ?? 0
        }
        
        navigationController?.popViewControllerAnimated(true)
    }
    
    func cancel() {
        // do nothing, edited variables will simply be deleted
        
        navigationController?.popViewControllerAnimated(true)
    }
    
    func nameDidChange(sender: UITextField) {
        timeZoneName = sender.text
    }
    
    func hideKeyboard() {
        view.endEditing(true)
    }
    
    // MARK: UITableViewDataSource
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 2
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        if section == 0 {
            return 1
        } else {
            return colors.count
        }
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        if indexPath.section == 0 {
            let cell = tableView.dequeueReusableCellWithIdentifier("TimeZoneNameCell") as? TimeZoneNameCell ?? TimeZoneNameCell(style: .Default, reuseIdentifier: "TimeZoneNameCell")
            cell.nameTextField.delegate = self
            cell.nameTextField.text = timeZoneName
            return cell
        } else {
            let cell = tableView.dequeueReusableCellWithIdentifier("TimeZoneColorCell") as? TimeZoneColorCell ?? TimeZoneColorCell(style: .Default, reuseIdentifier: "TimeZoneColorCell")
            
            let color = colors[indexPath.row]
            if timeZoneColorIndex == indexPath.row {
                cell.accessoryType = .Checkmark
                selectedIndexPath = indexPath
            }
            
            cell.colorNameLabel.text = String(color)    // Swift returns enum value's name as a string
            cell.colorMark.color = color.value
            return cell
        }
    }
    
    // MARK: UITableViewDelegate
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        if indexPath.section == 1 {
            if let selectedIndexPath = selectedIndexPath {
                tableView.cellForRowAtIndexPath(selectedIndexPath)?.accessoryType = .None
            }
            
            timeZoneColorIndex = indexPath.row
            tableView.cellForRowAtIndexPath(indexPath)?.accessoryType = .Checkmark
            selectedIndexPath = indexPath
            
            tableView.deselectRowAtIndexPath(indexPath, animated: true)
        }
    }
}

// MARK: UITextFieldDelegate
extension EditTimeZoneController: UITextFieldDelegate {
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        return textField.resignFirstResponder()
    }
}
